File: 3616-MakeArrayElementsEqualToZero/3616-MakeArrayElementsEqualToZero.cpp
// Last updated: 10/29/2025, 3:15:07 AM
class Solution {
public:
    int countValidSelections(vector<int>& nums) {
        int res = 0;
        int n = nums.size();
        // check for all zeros

        vector<int> left(n , 0);
        vector<int> right(n , 0);

        left[0] = nums[0];
        right[n - 1] = nums[n - 1];

        for(int i = 1; i < n; i++) {
            left[i] += left[i - 1] + nums[i];
        }
        for(int i = n - 2; i >= 0 ; i--) {
            right[i] += right[i + 1] + nums[i];
        }

        for(int i = 0; i < n ; i++) {
            if(nums[i] == 0) {
                if(left[i] == right[i]) res += 2;
                else if(abs(left[i] - right[i]) == 1)res += 1;
            }
        }
        
        return res;
        
    }
};

File: 3616-MakeArrayElementsEqualToZero/3616-MakeArrayElementsEqualToZero.java
// Last updated: 10/29/2025, 3:15:04 AM
class Solution {
    public int countValidSelections(int[] nums) {
        int n=nums.length, c=0;
        for(int i=0; i<n; i++)
        {
            if(nums[i] == 0)
            {
                int temp[]=Arrays.copyOf(nums, n);
                if(solve(temp, i, 1))
                    c++;
                temp=Arrays.copyOf(nums, n);
                if(solve(temp, i, -1))
                    c++;
            }
        }
        return c;
    }
    public boolean solve(int nums[], int curr, int dir)
    {
        if(curr<0 || curr>=nums.length)
        {
            for(int i=0; i<nums.length; i++)
                if(nums[i] != 0)
                    return false;
            return true;
        }

        if(nums[curr] == 0)
            return solve(nums, curr+dir, dir);
        else
        {
            nums[curr]--;
            if(dir == 1)
                dir=-1;
            else
                dir=1;
            return solve(nums, curr+dir, dir);
        }
    }
}